src/client/src/Stream.c

```diff
@@ -777,7 +777,11 @@ STATUS putFrame(PKinesisVideoStream pKinesisVideoStream, PFrame pFrame)
 
     // Set the last PutFrame time to current time
     currentTime = pKinesisVideoClient->clientCallbacks.getCurrentTimeFn(pKinesisVideoClient->clientCallbacks.customData);
-    pKinesisVideoStream->lastPutFrameTimestamp = currentTime;
+
+    if(pKinesisVideoStream->eofrFrame) {
```

Comment:
Should you move this below after the eofr is processed?

Reply:
addressed with LN823 comment